addons/base/views.py

```diff
@@ -693,6 +694,30 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
     return addon_view_file(auth, node, file_node, version)
 
 
+@must_be_valid_project(quickfiles_valid=True)
+@must_be_contributor_or_public
+def persistent_file_download(**kwargs):
+    id_or_guid = kwargs.get('fid_or_guid')
+    file = BaseFileNode.active.filter(_id=id_or_guid).first()
+    if not file:
+        guid = Guid.load(id_or_guid)
+        if guid:
+            file = guid.referent
+        else:
+            raise HTTPError(httplib.NOT_FOUND, data={
+                'message_short': 'File Not Found',
+                'message_long': 'The requested file could not be found.'
+            })
+
```

Comment:
What if the download link points to a folder rather than a file for osfstorage?
Below is the error I got when I have a folder inside a project and try to click the download link in the api
<img width="820" alt="screen shot 2018-01-24 at 11 49 00 am" src="https://user-images.githubusercontent.com/4974056/35345077-ad9df1a4-00fc-11e8-8af4-2cbf10236d61.png">

Reply:
Ah thanks for catching this. I shouldn't have links for folders at all.